### PR TITLE
Don't replace text outside of PGP armor in looking glass

### LINKED ISSUE #34: Don't replace text outside of PGP armor
In the looking glass, text outside of the PGP armor is ignored. So if I forward an email that has been clear-signed, users can't see the part outside of the forwarded mail.

### PR DESCRIPTION
**Pull request by diracdeltas**
Fix #34

### CODE REVIEW

**Comment by diracdeltas** [comment]
cc @adon-at-work for review.

**Comment by adon-at-work** [comment]
i lack an email to reproduce the described problem and verify the fix. but never mind, tried my best to review as much as i can. the code is of high quality, and it looks safe to merge. :)

**Comment by diracdeltas** [comment]
@adon-at-work To test, you can send yourself a email like:

```
this is an email header

-----BEGIN PGP MESSAGE----- 
Charset: UTF-8 
Version: Yahoo Mail End-To-End v0.0.5 

wf8AAAB+AzsR0bU37YgvEgIDBFT3UVHmWMkExQf7bNh35V04Yqbe8k20gXpt+wHF 
eU5LYxg+ZADK9xHJKxjsMl4ON0YMLmZvR2hcqs3qfkn98iUw9pIQsGUvED/S9pOA 
tqageNgPJSl+Vjmzpmzsow6HUd9Xy/fU7g2EKWHDRGS1IwKRwf8AAAB+AwSDtYBX 
9ZHuEgIDBCNyglLKnSC+A+4q5PVFMTStodGLI/vxwhe67zbfyExmYArdnhKTTSWE 
lXuFNkIHK2Cqmcndl8ePcpSkd+9CPmkwPAWXFR9DeFwfDP5wA4KqHS0Jagkhcdpo 
EaYgHu1tG6YxciiSPQc1cIDFRZC6uW0t0v8AAADbAU33fyv+XXv7ry3/YJ1G70j7 
4IVSTGnJ0woZCz9gD2SqiLgMp4M+pcymmrdMZnWIvDrjxqGB4S8oPQjoGwXiBWbZ 
le+1QuxD0wH6sA7Wymz3a2bM2IwExes/5onyvZTqcOS+fl6J/uoklxYNcNxfYNlI 
SdNs7EvbbFRYZPzCJG0k/DVEkCljIzdqsqnfKL7nUNNl2o9KVbi5Xv7e2zuVbofh 
utzwnBwKye/1HRgnnuOZxq0d84nQyBeIWpIpTmrR/slMS5h5jD85R0Yv+ZaSS059 
OcXVXZ+bjBvN 
=S/Jr 
-----END PGP MESSAGE----- 

this is an email footer
```

Before the patch, the expected result is that the entire email is replaced with the iframe, obscuring the header and footer text.

After the patch, only the PGP blob is replaced.

**Comment by diracdeltas** [comment]
> how would eliminating the use of opt_text affect functions like the following?
> https://github.com/yahoo/end-to-end/blob/2492829f5788f40eee4f20864dffcec9ba0447f5/src/javascript/crypto/e2e/extension/helper/e2ebind.js#L581

opt_text still gets passed around as a function parameter. It is only used to set the `original_text` attribute which can be useful for debugging.

**Comment by adon-at-work** [comment]
I tested by sending myself an email like you said, but the fix didn't work as expected.

File: src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js [hunk 1]
```diff
@@ -93,24 +86,20 @@ ui.GlassWrapper.prototype.installGlass = function(opt_callback) {
...
-    pgpMessage = this.originalText_;
-  } else {
-    pgpMessage =
-        e2e.openpgp.asciiArmor.extractPgpBlock(this.targetElem_.innerText);
-    surroundings = this.targetElem_.innerText.split(pgpMessage);
-  }
+
+  var pgpMessage =
+      e2e.openpgp.asciiArmor.extractPgpBlock(this.targetElem_.innerText);
+  var surroundings = this.targetElem_.innerText.split(pgpMessage);
```
Comment (adon-at-work) [inline src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js:92]: after debugging, problem found at `extractPgpBlock()` and `String.split()`

extractPgpBlock applies right trim to the pgp block and returns `pgpMessage`
https://github.com/yahoo/end-to-end/blob/master/src/javascript/crypto/e2e/openpgp/asciiarmor.js#L363

so `pgpMessage` can never be successful in splitting `this.targetElem_.innerText`, and as a result, the surronding texts are still missing.

Comment (adon-at-work) [inline src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js:92]: Notice that your example contains trailing space per line inside the pgp block

I drafted a secure email using the extension (and also tried to manually add some random prefix and suffix too). but the pgp block also contain trailing spaces.

Comment (diracdeltas) [inline src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js:92]: Weird. The example worked fine for me but now I have no idea why. :/

Comment (adon-at-work) [inline src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js:92]: i'm forwarding my email to you too for your examination (i operated my local key server).

Should I merge this PR as is? You may like to make a more resilient patch later.

Comment (diracdeltas) [inline src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js:92]: I haven't repro'ed your bug with the email you forwarded, so I guess this is good to merge for now. (But please open an issue with the example email and screenshot of the bug you're seeing.)

Comment (adon-at-work) [inline src/javascript/crypto/e2e/extension/ui/glass/glasswrapper.js:92]: sure.

**Pull request merged**